### PR TITLE
chore(deps): update all dependencies

### PR DESCRIPTION
**Pull request by DaniPopes**
Lots of duplicate dependencies when used as a dependency: https://t.me/ethers_rs/26283

I would also consider separating the SVM/Solc binaries from the SVM library into another crate to ease dep requirements (like clap, only used for the binary but it is also included as a dependency if imported by another crate).
I can open a follow-up PR with these changes if this is OK for you @roynalnaruto.

- ran `cargo update` + `cargo upgrade` 
- fixed minor breaking changes with `indicatif::ProgressBar`

### CODE REVIEW

**Comment by DaniPopes** [comment]
Hey @roynalnaruto, any updates? This would be very beneficial for any crates that use this one as a library. Thanks

File: svm-builds/build.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![allow(dead_code)]
```
Comment (roynalnaruto) [inline svm-builds/build.rs:1]: why is this needed?

File: svm-builds/Cargo.toml [hunk 2]
```diff
@@ -15,14 +15,14 @@ build = "build.rs"
 [build-dependencies]
 build_const = "0.2.2"
 svm = { package = "svm-rs", version = "0.2.22", path = "..", features = ["blocking"]}
-semver = { version = "1.0.4", default-features = false, features = ["std", "serde"] }
+semver = { version = "1.0.17", default-features = false, features = ["std", "serde"] }
```
Comment (roynalnaruto) [inline svm-builds/Cargo.toml:18]: I would prefer changing these to regular expressions, like in this case:
```suggestion
semver = { version = "^1.0.0", default-features = false, features = ["std", "serde"] }
```

Comment (roynalnaruto) [reply svm-builds/Cargo.toml:18]: And similarly in other places, so that we update the minor version in the PR and allow patches to be updated by `cargo update` in the future.

**Review by roynalnaruto — CHANGES_REQUESTED** [review]
We can update the dependency requirements to allow patch updates (using regular expressions instead of specific patch version). Other than that, thank you for contributing :)

File: svm-builds/build.rs (hunk 1 above)
Comment (DaniPopes) [reply svm-builds/build.rs:1]: The `generate` function is unused when the `_offline` feature is enabled, which generates a lot of warnings

File: svm-builds/Cargo.toml (hunk 2 above)
Comment (DaniPopes) [reply svm-builds/Cargo.toml:18]: `^1.0.0` is the default strategy, which is the same as `1.0.0`: <https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements>

I can set patch version to `0` or remove it if that's what you want

Comment (roynalnaruto) [reply svm-builds/Cargo.toml:18]: We can use `~1.0` in the above case, and similarly elsewhere then.

Comment (DaniPopes) [reply svm-builds/Cargo.toml:18]: Done in #81 since that PR will modify `Cargo.toml`s

**Comment by DaniPopes** [comment]
Closing in favor of #81
